zebra-network/src/protocol/external/codec.rs

```diff
@@ -113,6 +124,10 @@ impl Encoder for Codec {
             return Err(Parse("body length exceeded maximum size"));
         }
 
+        if self.builder.addr.is_some() {
+            metrics::counter!("bytes.written", (body.len() + HEADER_LEN) as u64, "addr" =>  self.builder.addr.unwrap().ip().to_string());
```

Comment:
rather than converting the addr to a string every time we increment this counter we might want to store the addr as a String and do the conversion up front just once.